### DISCUSSION TITLE
"let:" in NavBar

### DISCUSSION THREAD

**Discussion by deshartman** [themesberg/flowbite-svelte#494] [Q&A] [answered]
Hi,

Excuse the dumb question, but I have never seen the syntax used in Navbar, where the hidden and toggle variables are set using "let:" in HTML. Is this a HTML, CSS or Svelte specific syntax?

```
<Navbar let:hidden let:toggle>
  <NavHamburger on:click={toggle} />
  <NavUl {hidden}>
    <NavLi href="/" active={true}>Home</NavLi>
    <NavLi href="/about">About</NavLi>
  </NavUl>
</Navbar>
```

Would be great to just understand this

**Comment by shinokada** [comment, marked as answer]
@deshartman There are no dumb questions! Only questions you don't ask are dumb questions.
`let` is the Svelte's [`let` directive](https://svelte.dev/docs#template-syntax-slot-slot-key-value).
We use it when a parent component communicates with children components.

> Slots can be rendered zero or more times, and can pass values back to the parent using props. The parent exposes the values to the slot template using the let: directive.
> 
> The usual shorthand rules apply — let:item is equivalent to let:item={item}, and <slot {item}> is equivalent to <slot item={item}>.

**Reply by deshartman** [reply]
Perfect, thanks. I completely missed that in the Svelte docs.
